rust_src/crates/lisp/src/thread.rs

```diff
@@ -0,0 +1,25 @@
+//! Threading code
+
+use std::mem;
+
+use crate::{
+    buffer::LispBufferRef,
+    lisp::ExternalPtr,
+    remacs_sys::{current_thread as current_thread_pointer, thread_state},
+};
+
+pub type ThreadStateRef = ExternalPtr<thread_state>;
+
+pub struct ThreadState {}
+
+impl ThreadState {
+    pub fn current_buffer_unchecked() -> LispBufferRef {
```

Comment:
since this is `unchecked` do we want to mark it unsafe? What it is doing seems pretty unsafe to me.

Reply:
I'd say no. It doesn't help anyone. 

Usually marking a function `unsafe` communicates to the caller that they need to enforce a certain invariant or they will cause UB. You're not meant to tinker with the `current_thread_pointer` anyway, so safe rust will not cause UB in this function. 

The same goes for the other function.